src/self-update/README.md

```diff
@@ -0,0 +1,66 @@
+# Introduction
+
+That document specifies the feature self-update for thin-edge.
+
+The self-update allows a Cloud Operator to update one thin-edge version that is installed on a device to a newer version, via the Cloud connection. The feature is referred as self-update cause that update process is managed by thin-edge own's Software Management.
+
+# Requirements
+
+* Be default self-update shall use Debian Package Manager APT.
+  * NOTE: APT Package Manager provides a very basic rollback mechanism that still can fail (e.g. due to some unexpected device state or power-loss).
+* Device Owner shall be able to use a power-fail safe filesystem and transaction-safe package manager (instead of APT) - if the device support any.
+  * Therefore an extensible plugin shall be provided.
+* In thin-edge package's installation logic no external tools (e.g. grep, sed, ...) shall be used. Instead Rust code/libraries shall be used.
+  * This is to reduce external dependencies and so reduce risc for failures due missing tools or wrong version of tools.
+
+# Expected Pre-Conditions 
+
+The device and it's thereon installed thin-edge has to follow precondition below to allow successful self-update:
+
+1) The current (old) thin-edge on the device must be installed with usual thin-edge installation, or at leat as usual thin-edge installation would do. That is to assure new thin-edge version that is about to be installed has all external and internal files, folders and resources (e.g. config files) available.
```

Comment:
I don't understand what you mean by "or at leat as usual thin-edge installation would do".